homeassistant/components/light/limitlessled.py

```diff
@@ -271,128 +225,72 @@ def rgb_color(self):
     @property
     def supported_features(self):
         """Flag supported features."""
-        return SUPPORT_LIMITLESSLED_RGB
+        return self._supported
+
+    # pylint: disable=arguments-differ
+    @state(False)
+    def turn_off(self, transition_time, pipeline, **kwargs):
```

Comment:
Who calls this with the pipeline arg ? And why can't we store it on the instance?

Reply:
That's the `@state` decorator adding an argument, had me confused as well.

I do not want to rework further in this PR but it would probably be more readable to convert the decorator into a context manager. Or just write it out since it is no longer reused in three similar classes.